src/pcmg_supplement.c

```diff
@@ -0,0 +1,102 @@
+#include <petsc.h>
+#include <petscsys.h>
+#include <petscfix.h>
+#include <petscksp.h>
+#if ((PETSC_VERSION_MAJOR<=3) && (PETSC_VERSION_MINOR<=5))
+#include <petsc-private/fortranimpl.h>
+#include <petsc-private/pcimpl.h>
+#else
+#include <petsc/private/fortranimpl.h>
+#include <petsc/private/pcmgimpl.h>
+#endif
+
+#ifdef PETSC_USE_POINTER_CONVERSION
+extern void *PetscToPointer(void*);
+extern int PetscFromPointer(void *); 
+extern void PetscRmPointer(void*);
+#else
+#define PetscToPointer(a) (*(PetscFortranAddr *)(a))
+#define PetscFromPointer(a) (PetscFortranAddr)(a)
+#define PetscRmPointer(a)
+#endif
+
+#if ((PETSC_VERSION_MAJOR<=3) && (PETSC_VERSION_MINOR<=5))
+//This is taken from petsc-3.5/src/ksp/pc/impls/mg/mgimpl.h
+//There is no version of petsc/private/pcmgimpl.h that is includable in older
+//  versions of PETSc, so we just have to manually redefine the typdefs defined
+//  in mgimpl.h here.
```

Comment:
This is more of a question for @bkochuna, @bscollin, or @stimpsonsg, but would it be more appropriate to just not support some of these features if built with an old version of petsc?  It doesn't seem unreasonable to ask people to use a somewhat recent version of libraries if they want the latest features we have.  And duplicating petsc type definitions insides futility isn't ideal.

Reply:
Personally, I think it would be great to stop supporting PETSc 3.5 :)